packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -73,20 +72,17 @@ export function streamFromString(str: string): ReadableStream<Uint8Array> {
 export async function streamToString(
   stream: ReadableStream<Uint8Array>
 ): Promise<string> {
-  let buffer = ''
-
-  await stream
-    // Decode the streamed chunks to turn them into strings.
-    .pipeThrough(createDecodeTransformStream())
-    .pipeTo(
-      new WritableStream<string>({
-        write(chunk) {
-          buffer += chunk
-        },
-      })
-    )
+  const decoder = new TextDecoder()
+  let string = ''
+
+  // @ts-expect-error TypeScript gets this wrong (https://nodejs.org/api/webstreams.html#async-iteration)
+  for await (const chunk of stream) {
+    string += decoder.decode(chunk, { stream: true })
+  }
+
+  string += decoder.decode()
```

Comment:
Why is a last decode call needed?

Reply:
https://nodejs.org/api/util.html#textdecoderdecodeinput-options

> Decodes the input and returns a string. If options.stream is true, any incomplete byte sequences occurring at the end of the input are buffered internally and emitted after the next call to textDecoder.decode().

Reply:
Essentially, this makes sure everything is decoded, even if there are incomplete bytes.